src/main/java/de/tum/cit/aet/artemis/communication/web/SavedPostResource.java

```
@@ -0,0 +1,215 @@
+        return new ResponseEntity<>(postingList, null, HttpStatus.OK);
+    }
+
+    /**
+     * POST /saved-posts/{postId}/{type} : Create a new saved post
+     *
+     * @param postId post to save
+     * @param type   post type (post, answer)
+     * @return ResponseEntity with status 201 (Created) if successfully saved post,
+     *         or with status 400 (Bad Request) if the checks on type or post validity fail
+     */
+    @PostMapping("saved-posts/{postId}/{type}")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Void> savePost(@PathVariable Long postId, @PathVariable short type) {
+        log.debug("POST savePost invoked for post {}", postId);
+        long start = System.nanoTime();
+
+        if (savedPostService.isMaximumSavedPostsReached()) {
+            throw new BadRequestAlertException("The maximum amount of saved posts was reached.", ENTITY_NAME, "savedPostMaxReached");
+        }
+
+        var post = retrievePostingElseThrow(postId, type);
+
+        this.savedPostService.savePostForCurrentUser(post);
+
+        log.info("savePost took {}", TimeLogUtil.formatDurationFrom(start));
+        return new ResponseEntity<>(null, null, HttpStatus.CREATED);
+    }
+
+    /**
+     * DELETE /saved-posts/{postId}/{type} : Remove a saved post
+     *
+     * @param postId post to save
+     * @param type   post type (post, answer)
+     * @return ResponseEntity with status 204 (No content) if successfully deleted post,
+     *         or with status 400 (Bad Request) if the checks on type or post validity fail
+     */
+    @DeleteMapping("saved-posts/{postId}/{type}")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Void> deleteSavedPost(@PathVariable Long postId, @PathVariable short type) {
+        log.debug("DELETE deletePost invoked for post {}", postId);
+        long start = System.nanoTime();
+
+        var post = retrievePostingElseThrow(postId, type);
+
+        this.savedPostService.removeSavedPostForCurrentUser(post);
```

When the user tries to delete a post owned by another entity, this call silently return but neither in the logs nor in the client, the "action denied" event is shown. Have you considered notifying about it?

---

I added an exception here, so that the user gets a FORBIDDEN now if they try to delete a post that is not from them. 👍